lib/membrane_wav/serializer.ex

```diff
@@ -115,4 +146,32 @@ defmodule Membrane.WAV.Serializer do
       @data_length::32-little
     >>
   end
+
+  defp maybe_update_header_actions(%{update_header?: false}), do: []
+
+  defp maybe_update_header_actions(%{header_length: header_length, data_length: data_length}) do
+    case Code.ensure_compiled(Membrane.File.SeekEvent) do
+      {:module, seek_event} ->
+        # subtracting 8 bytes as `file_length` field doesn't include "RIFF" header and the field itself
+        file_length = header_length + data_length - 8
+
+        [
+          event: {:output, struct!(seek_event, position: @file_length_offset)},
+          buffer: {:output, %Buffer{payload: <<file_length::32-little>>}},
+          event: {:output, struct!(seek_event, position: @data_length_offset)},
+          buffer: {:output, %Buffer{payload: <<data_length::32-little>>}}
+        ]
+
+      {:error, reason} ->
+        require Membrane.Logger
+
+        Membrane.Logger.warn("""
+        Couldn't update WAV header as `Membrane.File.SeekEvent` module is not available (reason: #{inspect(reason)}).
+        You can use `Membrane.File.Sink` in your pipeline to correctly save the file or fix it later using
+        `Membrane.WAV.Postprocessing`.
+        """)
+
+        []
+    end
+  end
```

Comment:
This is invoked from an app, so `ensure_loaded?` should be enough. Also, if the check was moved outside, you could use the proper struct creation syntax and have compile-time check for the keys:

```suggestion
  if Code.ensure_loaded?(Membrane.File.SeekEvent) do
    defp maybe_update_header_actions(%{
           update_header?: true,
           header_length: header_length,
           data_length: data_length
         }) do
      # subtracting 8 bytes as `file_length` field doesn't include "RIFF" header and the field itself
      file_length = header_length + data_length - 8

      [
        event: {:output, %Membrane.File.SeekEvent{position: @file_length_offset}},
        buffer: {:output, %Buffer{payload: <<file_length::32-little>>}},
        event: {:output, %Membrane.File.SeekEvent{position: @data_length_offset}},
        buffer: {:output, %Buffer{payload: <<data_length::32-little>>}}
      ]
    end
  else
    defp maybe_update_header_actions(%{update_header?: true}) do
      # Suppress warnings
      _offset = @file_length_offset
      _offset = @data_length_offset

      raise "Unable to update WAV file header using seek events. [...]"
    end
  end
```

This can be verified by replacing `position` with `positions` (assuming file plugin is still a test-only dependency) will result in compilation error for `test` env and clean compilation on `dev`

Reply:
Surely that's way better!